code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/TokenUtils.java

```diff
@@ -284,4 +300,55 @@ public Account getOrCreateAccount(Identity gotIdentity, Set<Identity> identities
         }
         return account;
     }
+
+
+    public Token createToken(Set<Identity> identities, Account account) {
+
+        Identity user = getUser(identities);
+
+        if (user == null) {
+            throw new ClientVisibleException(ResponseCodes.UNAUTHORIZED);
+        }
+
+        account = getOrCreateAccount(user, identities, account);
+
+        if (account == null){
+            throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR, "FailedToGetAccount");
+        }
+
+        postAuthModification(account);
+
+        authDao.updateAccount(account, account.getName(), account.getKind(), user.getExternalId(), user
+                .getExternalIdType());
+
+        Map<String, Object> jsonData = new HashMap<>();
+        jsonData.put(TokenUtils.TOKEN, tokenType());
+        jsonData.put(TokenUtils.ACCOUNT_ID, user.getExternalId());
+        jsonData.put(TokenUtils.ID_LIST, identitiesToIdList(identities));
+        account = objectManager.reload(account);
```

Comment:
I think it would make sense to make authDao.updateAccount() to return an updated account. Would save you and extra DB call that you do by objectManager.reload(account)